docker/README.md

```diff
@@ -57,15 +57,15 @@ storage:
 Ok, now we are ready to run the Docker container! Use the following command: 
 
 ```
-docker run --rm -it -p 127.0.0.1:8192:8192 -p 127.0.0.1:4040-4050:4040-4050 -v `pwd`:/opt/config -v `pwd`/notebooks:/opt/notebooks polynote/polynote:latest --config /opt/config/config.yml
+docker run --rm -it -p 127.0.0.1:8192:8192 -p 127.0.0.1:4040-4050:4040-4050 -v `pwd`/config.yml:/opt/config/config.yml -v `pwd`/notebooks:/opt/polynote/notebooks/saved polynote/polynote:latest --config /opt/config/config.yml
 ```
 
 Let's go over what this command does. 
 
 - `-p 127.0.0.1:8192:8192` This binds `127.0.0.1:8192` on the host machine to port `8192` inside the container.
 - `-p 127.0.0.1:4040-4050:4040-4050` This binds a range of ports on the host machine so you can see the Spark UI. 
-- ``-v `pwd`:/opt/config`` This mounts the current directory (the one you just created) into the container at `/opt/config`
-- ``-v `pwd`/notebooks:/opt/notebooks`` This mounts the `./notebooks` directory on the host into the container at `/opt/notebooks`, so that any notebooks you create are saved back to the host. 
+- ``-v `pwd`/config.yml:/opt/config/config.yml`` This mounts the config file in the current directory (the one you just created) into the container at `/opt/config/config.yml`
+- ``-v `pwd`/notebooks:/opt/polynote/notebooks/saved`` This mounts the `./notebooks` directory on the host into the container at `/opt/polynote/notebooks/saved`, so that any notebooks you create are saved back to the host, and the example notebooks inside of `/opt/polynote/notebooks/` are still visible to you.
```

Comment:
@mathematicalmichael Are you testing this with the latest version and config as specified in the readme? I believe the readme in master should work fine if you're using `0.2.13` / `latest` release. Let me know if that's not the case.  

With the storage changes, everything in the root directory is persisted to the host volume aside from changes to the examples which are not mounted.

Reply:
The version I was using in my tests was corresponding to `0.2.13/latest` (`export BASE_IMAGE=polynote/polynote:latest`) 

I'm sorry, I don't think I follow you on the storage changes remark. The host volume is the folder on _my_ machine with `config` and `notebooks` in it, and the root directory is what, exactly? `/opt/polynote/`? Is there a desire to mount files other than the notebooks made in `saved` and `config.yml`? 

I also just double-checked my config and it only has the `listen` lines. I'll go ahead and retry with the newer config file just in case.

Reply:
just updated my config.yaml to have the new changes.
(really do consider adding this in by default to the docker folder)

I am unable to see examples
```
docker run --rm -it -e PYSPARK_ALLOW_INSECURE_GATEWAY=1 -p 127.0.0.1:8192:8192 -p 127.0.0.1:4040-4050:4040-4050 -v `pwd`/config.yml:/opt/config/config.yml -v `pwd`/notebooks:/opt/notebooks polynote-local --config /opt/config/config.yml
```

(I built `polynote-local`)

Reply:
No worries, I think I could definitely find a better way to explain this storage stuff. #634 is the PR with a few more details on how it works. 

I'm definitely able to see the examples on my end while following the current master instructions. Is it possible that you might need to `docker pull polynote/polynote:latest` before building `polynote-local` (your `latest` might be pointing to an older image)? Can you see the examples if you use, say, `polynote/polynote:0.2.13-2.11` along with the new config? If not, it might be a bug :(

In case it helps, here's what my `pwd` looks like:
```
├── config.yml
└── notebooks
    └── my\ first\ notebook.ipynb
```
And heres my run command: 
```
docker run --rm -it -p 127.0.0.1:8192:8192 -p 127.0.0.1:4040-4050:4040-4050 -v `pwd`/config.yml:/opt/config/config.yml -v `pwd`/notebooks:/opt/notebooks polynote/polynote:latest --config /opt/config/config.yml
```

* * * 

I'll try to explain the way these storage mounts work a little better. Let's break down the config (leaving aside the docker mounts for now):

```
storage:
  dir: /opt/notebooks
```

This part means that `/opt/notebooks` becomes the "root" directory you see in the UI. Thus, new notebooks that you make at the root of the UI will be written to `/opt/notebooks` - for example, if you created a notebook called `foo.ipynb`, it'd show up as just `foo.ipynb` in the UI and would be backed by `/opt/notebooks/foo.ipynb` on the filesystem. 

```
  mounts:
    examples:
      dir: examples
```

Here, the key `examples:` says to create a new "mount point" called `examples`, which is mounted relative to the notebook working directory (that is, relative to "root" as discussed in the previous paragraph). So in the UI, this looks like a directory named `examples`. 

The `dir: examples` part says that this mount point is backed by a directory which is also called `examples`. Being a relative path, it expands to `$POLYNOTE_WORKING_DIR/examples` which is `/opt/polynote/examples` in this case. So if you created a new notebook called `examples/foo.ipynb` it'd be persisted to `/opt/polynote/examples/foo.ipynb`.

Another way to think of these mount points is as symlinks. So think of it like this:

`storage: dir: /opt/notebooks` ==> `export $NOTEBOOK_DIR=/opt/notebooks`. 
`mounts: examples: dir: examples` ==> `ln -s $POLYNOTE_WORKING_DIR/examples $NOTEBOOK_DIR/examples`. 

And then imagine Polynote starts with `cd $NOTEBOOK_DIR`.  

Ok so let's come back to Docker. Now, if you use Docker to mount `/opt/notebooks` onto your host, all the notebooks you create in Polynote will be persisted to the host - _except_ for the notebooks in `examples`. 

Does that make sense? Let me know if there's anything that's still unclear - helps me make sure I can explain this clearly! 😄

Reply:
oh that's so helpful. I'm going to paraphrase and turn it into documentation as part of this PR. that explanation definitely needs to be in there in some form. I have done all sorts of mounting with docker before but I never quite had a config file so terse.

I have a project that I need to architect today at work but I'll come around to this at the end of the day and clean it up. thank you in advance for your patience, and thank you in retrospect for your responsiveness!

Reply:
Glad it was helpful, no problem! Thanks as well for sticking with this PR! 

One thing I think is important to clarify here is that all this storage stuff I'm referring to is a Polynote feature and nothing to do with Docker. 

It's a bit confusing because Polynote and Docker now both have a concept called "mounts", but they are not related.

The reason we care about this in relation to Docker is because Polynote's `Storage Mounts` tell Polynote where to look for files, and Docker mounts tell Docker which files to make available from the host, so as a user if I have files on my local machine I want Polynote to look at they need to be Docker-mounted into the container and then Polynote-mounted (using the config) so Polynote can find them.